### PR TITLE
Add bxt_lightstyle

### PR DESCRIPTION
**Pull request by khanghugo**
aka night vision

No need for mat_fullbright 1 and -75% fps if use this one instead. No need for tweaking gamma and brightness either. Most importantly, no need to restart the map.

### CODE REVIEW

**Review by YaLTeR — CHANGES_REQUESTED** [review]

File: src/modules/lightstyle.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+4: Mildy darker",
+);
+static BXT_LIGHTSTYLE_CUSTOM: CVar = CVar::new(
+    b"bxt_lightstyle_custom\0",
+    b"\0",
+    "\
+Custom controls. Takes precedence over preset when using bxt_lightstyle_apply.
+
+First value is effect. Second value is amount.
+E.g.: bxt_lightstyle_custom \"1 nomnomnom\".
```
[the remark this answers is not in the document]
Comment (khanghugo) [reply src/modules/lightstyle.rs:56]: This buifeature is pretty niche. First value is index of light. In renderfx, it lists all the styles. 0 is default lighting. Second value is the actual style where each character represents the brightness and loops for 0.1s.

https://twhl.info/wiki/page/Entity_Attribute:_Custom_Appearance

File: src/modules/lightstyle.rs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        cl_lightstyle[index as usize].map[..lightinfo.len()]
+            .copy_from_slice(lightinfo.as_slice_of().unwrap());
+        cl_lightstyle[index as usize].length = lightinfo.len() as i32;
+    }
+}
+
+pub fn on_cl_parse_lightstyle(marker: MainThreadMarker) {
+    // It is possible that the map has a preferred light style.
+    // Then, if we don't have any thing for our cvar, which is style is normal
+    // and no custom. THen we just don't do anything.
```
[the remark this answers is not in the document]
Comment (khanghugo) [reply src/modules/lightstyle.rs:119]: I have been wanting to add this but this PR keeps slipping of my mind. There is one problem, for the sake of simpler implementation, I would just restore the lighting of normal style (0).

**Comment by khanghugo** [comment]
I did a more testing and I see that a named light won't be overridden by this. But general lighting would be affected.

File: src/modules/lightstyle.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (YaLTeR) [reply src/modules/lightstyle.rs:56]: I see

File: src/modules/lightstyle.rs [hunk 3]
```diff
@@ -118,6 +131,10 @@ pub fn on_cl_parse_lightstyle(marker: MainThreadMarker) {
     // Then, if we don't have any thing for our cvar, which is style is normal
     // and no custom. THen we just don't do anything.
     if BXT_LIGHTSTYLE.as_u64(marker) != 0 || !BXT_LIGHTSTYLE_CUSTOM.to_string(marker).is_empty() {
+        let cl_lightstyle = &mut unsafe { *engine::cl_lightstyle.get(marker) };
+
+        // More often a map's default lightstyle will be empty.
+        *ORIGINAL_LIGHTSTYLE.borrow_mut(marker) = cl_lightstyle[0].map.to_vec();
         apply_from_cvars(marker);
```
Comment (YaLTeR) [inline src/modules/lightstyle.rs:138]: This is holding a `&mut` reference on lightstyle while it is modified in `apply_from_cvars`, which is UB. Limit the scope of the `&mut`.

Comment (khanghugo) [reply src/modules/lightstyle.rs]: Interesting.

**Comment by YaLTeR** [comment]
Thanks

**Pull request merged**